specification/hdinsight/resource-manager/Microsoft.HDInsight/stable/2018-06-01-preview/cluster.json

```diff
@@ -1522,6 +1529,21 @@
         "computeIsolationProperties": {
           "$ref": "#/definitions/ComputeIsolationProperties",
           "description": "The compute isolation properties."
+        },
+        "privateLinkConfigurations": {
+          "type": "array",
+          "description": "The private link configurations.",
+          "items": {
+            "$ref": "#/definitions/PrivateLinkConfiguration"
+          }
+        },
+        "privateEndpointConnections": {
+          "type": "array",
+          "description": "The list of private endpoint connections.",
+          "items": {
+            "$ref": "./privateEndpointConnections.json#/definitions/PrivateEndpointConnection"
```

Comment:
Can you instead just reference the v2 definitions at https://github.com/Azure/azure-rest-api-specs/blob/main/specification/common-types/resource-management/v2/privatelinks.json

Reply:
They also have one for 'PrivateEndpointConnectionListResult'

Reply:
For PrivateEndpointConnection related definitions, our service has an additional property “linkIdentifier”  in PrivateEndpointConnectionProperties
This is the main reason that I don’t reuse the definition in common-types/v1 or v2/privatelinks.json.  For this property if we miss this in our swagger we will get s360 issue.

Reply:
Would it be reasonable to use 'allOf' to inherit the common definition?

Reply:
> Would it be reasonable to use 'allOf' to inherit the common definition?

Hi @TimLovellSmith ,
Overall I think use `allOf` is good idea.
(a) `linkIdentifier` is in the `PrivateEndpointConnectionProperties` so we can only can use allOf to inherit this class, it means that we still need to define our own PrivateEndpointConnection, PrivateEndpointConnectionListResult
(b) Besides, I found that the `PrivateEndpointConnectionProvisioningState` in v2/privatelinks.json missed an enum "Updating". This also a problem.

I will try to find some one to ask whether we can add "Updating" in PrivateEndpointConnectionProvisioningState then try to use "allOf"